common/models/user.js

```diff
@@ -544,7 +544,11 @@ module.exports = function(User) {
     this.settings.ttl = this.settings.ttl || DEFAULT_TTL;
 
     UserModel.setter.password = function(plain) {
-      if (plain.indexOf('$2a$') === 0 && plain.length === 60) {
+      var indexOf = String.prototype.indexOf;
+      if (!plain) {
+        return;
+      }
+      if (indexOf.call(plain, '$2a$') === 0 && plain.length === 60) {
```

Comment:
I can remove the indexOf.call if it will help move things along.

The main thing I am worried about is users who have no passwords throwing and crashing.

The indexOf call was added as a bonus as address #1437